### ISSUE TITLE
Can't update cache: invalid peer certificate contents: invalid peer certificate: UnknownIssuer

### ISSUE THREAD

**Issue by freijon** [dbrgn/tealdeer#330]
When trying to run `tldr -u` I get the following error:

> Error: Could not update cache
> 
> Caused by:
>     0: error sending request for url (https://tldr.sh/assets/tldr.zip): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
>     1: error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
>     2: invalid peer certificate contents: invalid peer certificate: UnknownIssuer

I tried to download the ZIP file mentioned in the URL with `wget` and `curl` which works flawlessly.

What could be the issue here?

Version: tealdeer 1.6.1
OS: Linux

**Comment by sarg3nt** [comment]
I'm having the same problem.  I can `curl` it every time.
`curl  https://tldr.sh/assets/tldr.zip --output /tmp/tldr;`
I can sometimes get the update to work but most of the time it won't
```bash
tldr --update
Error: Could not update cache

Caused by:
    0: error sending request for url (https://tldr.sh/assets/tldr.zip): error trying to connect: dns error: failed to lookup address information: Name does not resolve
    1: error trying to connect: dns error: failed to lookup address information: Name does not resolve
    2: dns error: failed to lookup address information: Name does not resolve
    3: failed to lookup address information: Name does not resolve
```
This is during a container build or from within the built container but is also true form my Ubuntu Linux WSL instance.
NOTE: We are behind a corporate proxy.  Standard proxy vars are set including both uppercase and lowercase versions.  No proxy authentication.

**Comment by sarg3nt** [comment]
> About "we are behind a proxy": Are the two of you in the same company / network?

Sorry, no, I don't think so.  Not sure why I used the royal we there.  :) 

Github was having some service outages this morning, so that might have contributed.
I can now repeatedly update in a Linux box and I can run update the first time in a dev container based on rocky 9 however, running update a second time gives  `failed to lookup address information: Name does not resolve` still.

**Comment by freijon** [comment]
But the "we" was not entirely false - I'm also behind a corporate proxy :) At least on the machine that has the issue. I will rebuild `tealdeer` with native-roots, webpki-roots and native-tls enabled after my holidays. But I'm wondering - Why are those features not enabled by default?

**Comment by niklasmohrin** [comment]
To build, exactly one of these features must be enabled. They map to equivalent features of the networking library we use. My current suspicion is that your builds have bundled certificates that are now somehow wrong / outdated, so a fresh build, depending on the feature, would use current system certificates or bundle a fresh set of certificates.

**Comment by niklasmohrin** [comment]
@sarg3nt your problem is unrelated to the certifcates, in your case DNS lookup fails. The first time / second time discrepancy is odd, not really sure what to make of that. In general, my suspicion would be that your system has some special DNS settings that our networking library is not picking up on. This would also explain why curl works. It doesn't really explain why tealdeer works sometimes 🤔 I am not sure right now how we should best pursue a resolution of this problem as it probably boils down to a library issue. I will come back to this later

**Comment by freijon** [comment]
I'm on Gentoo, so each time I reinstall the package it gets built from source (from your release tarball). I tried that already with no effect. The only thing left to try seems to be playing with some networking features.

**Comment by niklasmohrin** [comment]
@freijon Can you share the exact steps of your build?

**Comment by freijon** [comment]
Here you find the build recipe of the Gentoo tealdeer package: https://github.com/gentoo/guru/blob/master/app-misc/tealdeer/tealdeer-1.6.1.ebuild

Only the default features are used, nothing special

**Comment by freijon** [comment]
I just built the package with `webpki-roots` - Same error.
Then I tried `cargo build --release --features reqwest/native-tls --features native-roots` - Success!

Can we do this by default?

**Comment by niklasmohrin** [comment]
Hm, this seems like an odd selection of features. The first is what you get if you build tealdeer with `native-tls` while the former uses non-native tls (rustls) and I think reqwest will just pick one of the two (this is also why you have to compile tealdeer with exactly one of the three feature flags).

I think that even though you rebuild the package on Gentoo, it will use our lockfile that maybe contains outdated certificates? By running `cargo build` yourself (without `--locked`), you avoid the lockfile. This still leaves the question why your own build with `webpki-roots` failed... Maybe the dependency was then cached and your build reused it? Not entirely sure to be honest. I would also expect the default feature, which is `native-roots`, to keep working since that is their entire purpose :thinking:

**Comment by niklasmohrin** [comment]
@sarg3nt Did your problem ever re-occur?

**Comment by sarg3nt** [comment]
> @sarg3nt Did your problem ever re-occur?

Nope.

**Comment by sarg3nt** [comment]
I think we can close this as it seems to be working fine now.
@freijon are you still experiencing this?  If not, please close with comment.

**Comment by nick-s-b** [comment]
I just installed `tealdeer` on a fresh Arch install and  it fails every time with the exact same message:

````
$ tldr tldr 
Error: Page cache not found. Please run `tldr --update` to download the cache.

$ tldr --update
Successfully created cache directory path `/home/user/.cache/tealdeer`.
Error: Could not update cache

Caused by:
    0: error sending request for url (https://tldr.sh/assets/tldr.zip): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
    1: error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
    2: invalid peer certificate contents: invalid peer certificate: UnknownIssuer

$ curl -O https://tldr.sh/assets/tldr.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 8999k  100 8999k    0     0  30.5M      0 --:--:-- --:--:-- --:--:-- 30.7M
````

Not sure what's going on. It looks like [a bunch of Rust projects](https://www.google.com/search?q=%220%3A+error+sending+request+for+url%22+%22error+trying+to+connect%3A+invalid+peer+certificate+contents%3A+invalid+peer+certificate%3A+UnknownIssuer%22) are impacted by the same bug.

I'm not sure what to do next...

**Comment by niklasmohrin** [comment]
@nick-s-b That's of course annoying. To get tealdeer to work for now, you can unzip the `tldr.zip` manually so that the directory structure is something like
```plain
/home/user
 .cache
  tealdeer
   tldr-pages
    pages
    pages.ar
    pages.bn
    ...
```

If you don't care much about installing tealdeer from the Arch repositories vs. other methods, you could try installing with `cargo` or `nix`. In particular, using `cargo` with `--no-default-features --features native-tls` could work.

**Comment by niklasmohrin** [comment]
@dbrgn I think we need to think about the whole TLS story again, maybe we should change the default to use native TLS. I also saw that reqwest allows multiple backends that can then be chosen with methods on `ClientBuilder`: https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.use_rustls_tls Maybe we should include multiple methods in our builds and allow configuring the used one through the config, so that old builds can eventually make a switch to native TLS if rustls stops working. What do you think?

**Comment by niklasmohrin** [comment]
(going with the google result list from @nick-s-b, the issue seems to be related to rustls and the webpki crate it uses; I suppose going with native tls as the default is a reasonable move)

**Comment by nick-s-b** [comment]
@niklasmohrin thank you! The manual intervention worked ;). 

Do you think this could be related to IPv6? I have IPv6 disabled. I remember long time ago running into an issue with `fish` that would fail to bring up its internal conf local web server without IPv6 even though it used IPv4? `fish` team fixed this and now it doesn't care if you have IPv6 disabled.

**Comment by niklasmohrin** [comment]
I read a bit of what appears to me as the relevant fish code and issues, I think it is unrelated. They are trying to listen on an interface and prefer ipv6, but sometimes fail to make the correct choice. We are only trying to connect to another device. In contrast to fish, we don't decide whether to use ipv4 or ipv6 ourselves and let another library handle it.

From what I can tell, this is a bug in an underlying library with parsing the TLS certificate used by tldr.sh. It could of course be that ipv6 plays a subtle role in that bug, but I have no idea. It seems to me that this bug is happening on a higher level in the networking stack.

**Comment by nick-s-b** [comment]
@niklasmohrin OK, got it! Thank you for looking into this.

**Comment by niklasmohrin** [comment]
Sure, hope we can eliminate this source of bugs some time soon, or at least include a workaround

**Comment by dbrgn** [comment]
@nick-s-b are you in a home network, or in a company or university/school? To me, this looks like TLS interception in your network.

Tealdeer uses the certificate roots from WebPKI by default and ignores system certificates: https://github.com/rustls/webpki-roots This means that if you installed a custom TLS certificate root in your system, that will be ignored. However, the only way how a custom ceritificate root could be used by the `tldr.sh` domain is if some system in your network intercepts the request, does a man-in-the-middle and replaces the certificate.

> Not sure what's going on. It looks like [a bunch of Rust projects](https://www.google.com/search?q=%220%3A+error+sending+request+for+url%22+%22error+trying+to+connect%3A+invalid+peer+certificate+contents%3A+invalid+peer+certificate%3A+UnknownIssuer%22) are impacted by the same bug.

The result page only shows two different projects: railwayapp and deno.

**Comment by dbrgn** [comment]
> @dbrgn I think we need to think about the whole TLS story again, maybe we should change the default to use native TLS.

Given that the only damage an attacker could do is modifying the downloaded tldr pages, that might be a good default. From a security perspective, using WebPKI roots is better because noone can tamper with your encrypted connections, but on the other hand if your system trusts those roots, then why shouldn't tealdeer?

I wouldn't be opposed to a switch to native roots.

**Comment by nick-s-b** [comment]
@dbrgn Hi, I was connecting from home. It's my home PC with a cable internet connection.  I live in a "free" country. While it's certainly possible I'm getting MIM attacked, I think it's unlikely. But who knows... maybe I am. I'm not sure how to check and I don't see SSL errors when I visit emailsites or my bank site.

Here's some info from OpenSSL:

<details>
  <summary>OpenSSL cert output</summary>
  $ openssl s_client -connect tldr.sh:443 -showcerts
Connecting to 172.64.80.1
CONNECTED(00000003)
depth=2 C=US, O=Google Trust Services LLC, CN=GTS Root R4
verify return:1
depth=1 C=US, O=Google Trust Services, CN=WE1
verify return:1
depth=0 CN=tldr.sh
verify return:1
---
Certificate chain
 0 s:CN=tldr.sh
   i:C=US, O=Google Trust Services, CN=WE1
   a:PKEY: id-ecPublicKey, 256 (bit); sigalg: ecdsa-with-SHA256
   v:NotBefore: Aug 21 01:49:24 2024 GMT; NotAfter: Nov 19 01:49:23 2024 GMT
-----BEGIN CERTIFICATE-----
MIIDlzCCAz6gAwIBAgIQEhJC/1eDp1YR3PffBjyZazAKBggqhkjOPQQDAjA7MQsw
CQYDVQQGEwJVUzEeMBwGA1UEChMVR29vZ2xlIFRydXN0IFNlcnZpY2VzMQwwCgYD
VQQDEwNXRTEwHhcNMjQwODIxMDE0OTI0WhcNMjQxMTE5MDE0OTIzWjASMRAwDgYD
VQQDEwd0bGRyLnNoMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEkPE75w8r6Xq6
0Wez2lOYFAXE7P3X6ChdZgUagqWCNB+LQlbQ34eGK1WNOPY8UvkOT/I5BtAcxpci
9DVtHlO2/KOCAkswggJHMA4GA1UdDwEB/wQEAwIHgDATBgNVHSUEDDAKBggrBgEF
BQcDATAMBgNVHRMBAf8EAjAAMB0GA1UdDgQWBBQ5yu1LY1X+mEagqgbHiX9Nh6yv
4TAfBgNVHSMEGDAWgBSQd5I1Z8T/qMyp5nvZgHl7zJP5ODBeBggrBgEFBQcBAQRS
MFAwJwYIKwYBBQUHMAGGG2h0dHA6Ly9vLnBraS5nb29nL3Mvd2UxL0VoSTAlBggr
BgEFBQcwAoYZaHR0cDovL2kucGtpLmdvb2cvd2UxLmNydDAdBgNVHREEFjAUggd0
bGRyLnNoggkqLnRsZHIuc2gwEwYDVR0gBAwwCjAIBgZngQwBAgEwNgYDVR0fBC8w
LTAroCmgJ4YlaHR0cDovL2MucGtpLmdvb2cvd2UxL2h4V2ZlVmVjX1I0LmNybDCC
AQQGCisGAQQB1nkCBAIEgfUEgfIA8AB3AO7N0GTV2xrOxVy3nbTNE6Iyh0Z8vOze
w1FIWUZxH7WbAAABkXLWydkAAAQDAEgwRgIhAOcpwVCHw0SUsGYGN669SnRAxwJZ
17q2zxaWByMEmnlFAiEAxR8EJe6LqizDUgXfN8aC4Vov3f9K4i1fVqGhoTkkS3MA
dQDatr9rP7W2Ip+bwrtca+hwkXFsu1GEhTS9pD0wSNf7qwAAAZFy1soRAAAEAwBG
MEQCIBh9OPVp3e8gd4v36MDZUuUB3nrQhZbkA4q9RSQGyszBAiA7pRFJdXMd0F+y
omScNlbrBh6CP7OJqCjkH21nkkUTRzAKBggqhkjOPQQDAgNHADBEAiBKaXnab+7G
SQo2pqevKZi/P9I79pMnh5D22d9BSQsHqwIgIB0dWVZll6YkrRBykwrJe1HvstED
uf3H3EHJEJHadms=
-----END CERTIFICATE-----
 1 s:C=US, O=Google Trust Services, CN=WE1
   i:C=US, O=Google Trust Services LLC, CN=GTS Root R4
   a:PKEY: id-ecPublicKey, 256 (bit); sigalg: ecdsa-with-SHA384
   v:NotBefore: Dec 13 09:00:00 2023 GMT; NotAfter: Feb 20 14:00:00 2029 GMT
-----BEGIN CERTIFICATE-----
MIICnzCCAiWgAwIBAgIQf/MZd5csIkp2FV0TttaF4zAKBggqhkjOPQQDAzBHMQsw
CQYDVQQGEwJVUzEiMCAGA1UEChMZR29vZ2xlIFRydXN0IFNlcnZpY2VzIExMQzEU
MBIGA1UEAxMLR1RTIFJvb3QgUjQwHhcNMjMxMjEzMDkwMDAwWhcNMjkwMjIwMTQw
MDAwWjA7MQswCQYDVQQGEwJVUzEeMBwGA1UEChMVR29vZ2xlIFRydXN0IFNlcnZp
Y2VzMQwwCgYDVQQDEwNXRTEwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAARvzTr+
Z1dHTCEDhUDCR127WEcPQMFcF4XGGTfn1XzthkubgdnXGhOlCgP4mMTG6J7/EFmP
LCaY9eYmJbsPAvpWo4H+MIH7MA4GA1UdDwEB/wQEAwIBhjAdBgNVHSUEFjAUBggr
BgEFBQcDAQYIKwYBBQUHAwIwEgYDVR0TAQH/BAgwBgEB/wIBADAdBgNVHQ4EFgQU
kHeSNWfE/6jMqeZ72YB5e8yT+TgwHwYDVR0jBBgwFoAUgEzW63T/STaj1dj8tT7F
avCUHYwwNAYIKwYBBQUHAQEEKDAmMCQGCCsGAQUFBzAChhhodHRwOi8vaS5wa2ku
Z29vZy9yNC5jcnQwKwYDVR0fBCQwIjAgoB6gHIYaaHR0cDovL2MucGtpLmdvb2cv
ci9yNC5jcmwwEwYDVR0gBAwwCjAIBgZngQwBAgEwCgYIKoZIzj0EAwMDaAAwZQIx
AOcCq1HW90OVznX+0RGU1cxAQXomvtgM8zItPZCuFQ8jSBJSjz5keROv9aYsAm5V
sQIwJonMaAFi54mrfhfoFNZEfuNMSQ6/bIBiNLiyoX46FohQvKeIoJ99cx7sUkFN
7uJW
-----END CERTIFICATE-----
 2 s:C=US, O=Google Trust Services LLC, CN=GTS Root R4
   i:C=BE, O=GlobalSign nv-sa, OU=Root CA, CN=GlobalSign Root CA
   a:PKEY: id-ecPublicKey, 384 (bit); sigalg: RSA-SHA256
   v:NotBefore: Nov 15 03:43:21 2023 GMT; NotAfter: Jan 28 00:00:42 2028 GMT
-----BEGIN CERTIFICATE-----
MIIDejCCAmKgAwIBAgIQf+UwvzMTQ77dghYQST2KGzANBgkqhkiG9w0BAQsFADBX
MQswCQYDVQQGEwJCRTEZMBcGA1UEChMQR2xvYmFsU2lnbiBudi1zYTEQMA4GA1UE
CxMHUm9vdCBDQTEbMBkGA1UEAxMSR2xvYmFsU2lnbiBSb290IENBMB4XDTIzMTEx
NTAzNDMyMVoXDTI4MDEyODAwMDA0MlowRzELMAkGA1UEBhMCVVMxIjAgBgNVBAoT
GUdvb2dsZSBUcnVzdCBTZXJ2aWNlcyBMTEMxFDASBgNVBAMTC0dUUyBSb290IFI0
MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAE83Rzp2iLYK5DuDXFgTB7S0md+8Fhzube
Rr1r1WEYNa5A3XP3iZEwWus87oV8okB2O6nGuEfYKueSkWpz6bFyOZ8pn6KY019e
WIZlD6GEZQbR3IvJx3PIjGov5cSr0R2Ko4H/MIH8MA4GA1UdDwEB/wQEAwIBhjAd
BgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwDwYDVR0TAQH/BAUwAwEB/zAd
BgNVHQ4EFgQUgEzW63T/STaj1dj8tT7FavCUHYwwHwYDVR0jBBgwFoAUYHtmGkUN
l8qJUC99BM00qP/8/UswNgYIKwYBBQUHAQEEKjAoMCYGCCsGAQUFBzAChhpodHRw
Oi8vaS5wa2kuZ29vZy9nc3IxLmNydDAtBgNVHR8EJjAkMCKgIKAehhxodHRwOi8v
Yy5wa2kuZ29vZy9yL2dzcjEuY3JsMBMGA1UdIAQMMAowCAYGZ4EMAQIBMA0GCSqG
SIb3DQEBCwUAA4IBAQAYQrsPBtYDh5bjP2OBDwmkoWhIDDkic574y04tfzHpn+cJ
odI2D4SseesQ6bDrarZ7C30ddLibZatoKiws3UL9xnELz4ct92vID24FfVbiI1hY
+SW6FoVHkNeWIP0GCbaM4C6uVdF5dTUsMVs/ZbzNnIdCp5Gxmx5ejvEau8otR/Cs
kGN+hr/W5GvT1tMBjgWKZ1i4//emhA1JG1BbPzoLJQvyEotc03lXjTaCzv8mEbep
8RqZ7a2CPsgRbuvTPBwcOMBBmuFeU88+FSBX6+7iP0il8b4Z0QFqIwwMHfs/L6K1
vepuoxtGzi4CZ68zJpiq1UvSqTbFJjtbD4seiMHl
-----END CERTIFICATE-----
---
Server certificate
subject=CN=tldr.sh
issuer=C=US, O=Google Trust Services, CN=WE1
---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: ECDSA
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 2812 bytes and written 395 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 256 bit
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: 6C64A5B3301EA1F67C342A90FA9142D1DC449FAF4C86BEC973CD0B52775E150B
    Session-ID-ctx: 
    Resumption PSK: A5D40599F6E30352DA20F50E7DBC4E47ED649D0DCCA5C3D6B9A8B47297ACC513CAFBA0AAB5857C9F3C6B8CF23C693FD4
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 64800 (seconds)
    TLS session ticket:
    0000 - 4e 1a 4c be d8 fe c4 79-30 a3 0b e6 96 93 99 b2   N.L....y0.......
    0010 - 6e 2f 29 c5 60 aa d2 e4-ad 90 aa dd 0c c3 43 24   n/).`.........C$
    0020 - 6e ac c0 57 43 5f 5a 6b-96 b9 1a e9 0e 03 df e1   n..WC_Zk........
    0030 - 67 e0 cf 4c 44 f8 ea 03-ff 7c 0f bd 7a c2 8c db   g..LD....|..z...
    0040 - fd 50 d6 15 6e 7a fa 0f-f0 18 61 53 f2 e6 ca d1   .P..nz....aS....
    0050 - ac 05 41 64 4d 32 b9 37-c1 a3 e7 f5 fb 99 62 31   ..AdM2.7......b1
    0060 - f4 f5 5b 3d 37 4b 95 a4-46 f7 12 b9 16 f3 a2 36   ..[=7K..F......6
    0070 - a4 82 3d 0b 2d 17 29 98-66 37 07 0b c6 d5 25 1e   ..=.-.).f7....%.
    0080 - 16 84 78 25 2c 42 ca e3-3d e4 1b a9 b3 0d 03 7c   ..x%,B..=......|
    0090 - 7c eb d9 59 f5 63 26 f4-05 64 e5 50 15 b8 bf 9a   |..Y.c&..d.P....
    00a0 - 00 a2 f1 05 3f c7 35 ef-b1 23 90 fd a5 03 28 8c   ....?.5..#....(.
    00b0 - 6e 32 6a 87 9c 32 3b 6c-fc 49 e1 f2 f1 49 87 47   n2j..2;l.I...I.G

    Start Time: 1727913484
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: 0A39E8EFFA38E94D9210A3BCEADEC8B08EA6DF331E426DE7F8F830E8E39A91F3
    Session-ID-ctx: 
    Resumption PSK: DAF1F96A5E819191EF7E42A281FAA7B8A939FBA1A712C241CC1310C78ABE94C3277C9216BD42280ADDDDC9EB77481724
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 64800 (seconds)
    TLS session ticket:
    0000 - 4e 1a 4c be d8 fe c4 79-30 a3 0b e6 96 93 99 b2   N.L....y0.......
    0010 - 1f e6 4d e1 11 2f 55 14-a5 5b 2e 6d 66 45 5b b6   ..M../U..[.mfE[.
    0020 - dd e8 fb fe bf 2d 03 06-1b 96 ae 58 d0 57 60 82   .....-.....X.W`.
    0030 - c8 45 1d a2 97 b3 3d 39-87 9d 0f 8c 68 0e f0 22   .E....=9....h.."
    0040 - 2c 22 8c 38 91 bb 3f d5-04 9d 8e 03 55 c6 d1 39   ,".8..?.....U..9
    0050 - 70 ac 3a 74 34 0b 39 d3-19 a8 bd 8d 88 c2 56 92   p.:t4.9.......V.
    0060 - 37 a4 25 09 ce 68 8d 31-63 48 76 fb 74 6c 8d 65   7.%..h.1cHv.tl.e
    0070 - c9 5f 86 30 c7 e8 50 b1-e2 78 fa 86 67 d2 62 1c   ._.0..P..x..g.b.
    0080 - 1a b0 6f 04 27 0d e0 47-8d df 9f aa 74 c3 bf 96   ..o.'..G....t...
    0090 - 5e 53 82 63 68 cd 41 c4-38 e7 34 0c 9d 83 fd 61   ^S.ch.A.8.4....a
    00a0 - 0e e9 f4 ea 49 66 59 7f-41 6e 8f d8 38 5c 34 7f   ....IfY.An..8\4.
    00b0 - 59 47 7b 4f 7a 45 eb 46-a8 f9 9a e3 4f 56 69 0b   YG{OzE.F....OVi.

    Start Time: 1727913484
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
closed
  
</details>

And here is curl:

<details>
  <summary>Curl output:</summary>
  $ curl --verbose -O https://tldr.sh/assets/tldr.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Host tldr.sh:443 was resolved.
* IPv6: (none)
* IPv4: 172.64.80.1
*   Trying 172.64.80.1:443...
* ALPN: curl offers h2,http/1.1
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
*  CAfile: /usr/share/curl/ca-bundle.crt
*  CApath: none
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
{ [19 bytes data]
* TLSv1.3 (IN), TLS handshake, Certificate (11):
{ [2515 bytes data]
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
{ [80 bytes data]
* TLSv1.3 (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.3 (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384 / x25519 / id-ecPublicKey
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=tldr.sh
*  start date: Aug 21 01:49:24 2024 GMT
*  expire date: Nov 19 01:49:23 2024 GMT
*  subjectAltName: host "tldr.sh" matched cert's "tldr.sh"
*  issuer: C=US; O=Google Trust Services; CN=WE1
*  SSL certificate verify ok.
*   Certificate level 0: Public key type EC/prime256v1 (256/128 Bits/secBits), signed using ecdsa-with-SHA256
*   Certificate level 1: Public key type EC/prime256v1 (256/128 Bits/secBits), signed using ecdsa-with-SHA384
*   Certificate level 2: Public key type EC/secp384r1 (384/192 Bits/secBits), signed using ecdsa-with-SHA384
} [5 bytes data]
* Connected to tldr.sh (172.64.80.1) port 443
* using HTTP/2
* [HTTP/2] [1] OPENED stream for https://tldr.sh/assets/tldr.zip
* [HTTP/2] [1] [:method: GET]
* [HTTP/2] [1] [:scheme: https]
* [HTTP/2] [1] [:authority: tldr.sh]
* [HTTP/2] [1] [:path: /assets/tldr.zip]
* [HTTP/2] [1] [user-agent: curl/8.10.1]
* [HTTP/2] [1] [accept: */*]
} [5 bytes data]
> GET /assets/tldr.zip HTTP/2
> Host: tldr.sh
> User-Agent: curl/8.10.1
> Accept: */*
> 
* Request completely sent off
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [230 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [230 bytes data]
< HTTP/2 200 
< date: Wed, 02 Oct 2024 23:53:51 GMT
< content-type: application/zip
< content-length: 9226008
< last-modified: Wed, 02 Oct 2024 18:55:10 GMT
< access-control-allow-origin: *
< etag: "66fd970e-8cc718"
< expires: Wed, 02 Oct 2024 22:49:28 GMT
< cache-control: max-age=14400
< x-proxy-cache: MISS
< x-github-request-id: 3B40:38EED5:1063E1:119CE5:66FD9968
< via: 1.1 varnish
< x-served-by: cache-yyz4575-YYZ
< x-cache: MISS
< x-cache-hits: 0
< x-timer: S1727895912.258398,VS0,VE174
< vary: Accept-Encoding
< x-fastly-request-id: 5304ca2cbd2ea139b6e9d477ba484c6d92cf66b9
< cf-cache-status: REVALIDATED
< accept-ranges: bytes
< report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=vp0pqxBUmLjOeJrO%2BKBW0m5zGj%2B4w1Ne9LaBN45BTpZ%2BmIcA%2F%2Fkm9LuXTUkW2Vs7y4uktrqWEAJTjuNViBut2m9ksA83f7ZUTbYLd%2BccbHuCeyGLia%2F98vrx"}],"group":"cf-nel","max_age":604800}
< nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
< strict-transport-security: max-age=15552000
< x-content-type-options: nosniff
< server: cloudflare
< cf-ray: 8cc89d3ffc45ac4c-YYZ
< 
{ [5 bytes data]
100 9009k  100 9009k    0     0  29.7M      0 --:--:-- --:--:-- --:--:-- 29.9M
* Connection #0 to host tldr.sh left intact

  
  
</details>

**Comment by dbrgn** [comment]
Oh, that is interesting. It's also possible that version 1.6.1 of tealdeer (released 2 years ago) does not yet include some of the newer trust roots. The question is why it wouldn't affect all users in that case... However, it's possible that Cloudflare (where the files seem to be hosted) doesn't use the same certificate for all requests.

@nick-s-b yesterday we released version 1.7.0 of tealdeer, with updated roots. The package is already available in the regular ArchLinux repositories. Could you update to that version and test if it works?

**Comment by nick-s-b** [comment]
@dbrgn It works perfectly! Thank you!

````
$ tldr --version
tealdeer 1.7.0

$ tldr --update
Successfully updated cache.
````
